independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java

```diff
@@ -150,39 +167,88 @@ public void execute() throws MojoExecutionException {
-            extObject.writeTo(bw, WriterConfig.PRETTY_PRINT);
+
+        try (BufferedWriter bw = Files
+                .newBufferedWriter(output.resolve(BootstrapConstants.EXTENSION_PROPS_JSON_FILE_NAME))) {
+            bw.write(mapper.writer(prettyPrinter).writeValueAsString(extObject));
         } catch (IOException e) {
-            throw new MojoExecutionException("Failed to persist " + output.resolve(BootstrapConstants.EXTENSION_PROPS_JSON_FILE_NAME), e);
+            throw new MojoExecutionException(
+                    "Failed to persist " + output.resolve(BootstrapConstants.EXTENSION_PROPS_JSON_FILE_NAME), e);
         }
     }
 
-	private void transformLegacyToNew(final Path output, JsonObject extObject) throws MojoExecutionException {
-		JsonObject metadata = null;
-        
-        JsonValue mvalue = extObject.get("metadata");
-        if(mvalue!=null && mvalue.isObject()) {
-        	metadata = mvalue.asObject();
+    private void transformLegacyToNew(final Path output, ObjectNode extObject, ObjectMapper mapper)
+            throws MojoExecutionException {
+        ObjectNode metadata = null;
+
+        // Note: groupId and artifactId shouldn't normally be in the source json but
+        // just putting it
+        // here for completenes
+        if (extObject.get("groupId") != null) {
+            extObject.set(GROUP_ID, extObject.get("groupId"));
+            extObject.remove("groupId");
+        }
+
+        if (extObject.get("artifactId") != null) {
+            extObject.set(ARTIFACT_ID, extObject.get("artifactId"));
+            extObject.remove("artifactId");
+        }
+
+        JsonNode mvalue = extObject.get("metadata");
+        if (mvalue != null && mvalue.isObject()) {
+            metadata = (ObjectNode) mvalue;
         } else {
-        	metadata = new JsonObject();
-        }
-        
-        if(extObject.get("labels") != null) {
-        	metadata.add("keywords", extObject.get("labels"));
-        	extObject.remove("labels");
-        }
-        
-        extObject.set("metadata", metadata );
-        
-	}
-    
+            metadata = mapper.createObjectNode();
+        }
+
+        if (extObject.get("labels") != null) {
+            metadata.set("keywords", extObject.get("labels"));
+            extObject.remove("labels");
+        }
+
+        if (extObject.get("guide") != null) {
+            metadata.set("guide", extObject.get("guide"));
+            extObject.remove("guide");
+        }
+
+        if (extObject.get("shortName") != null) {
+            metadata.set("short-name", extObject.get("shortName"));
+            extObject.remove("shortName");
+        }
+
+        extObject.set("metadata", metadata);
+
+        // TODO: remove before going to master
+        Path source = output
+                .resolve("../../../src/main/resources/META-INF/");
+        System.out.println("Try to save " + source);
```

Comment:
Forgot to remove?

Reply:
it’s the code generating the update json and now yaml. will be removed before final merge.

Reply:
removed it now so we can get this change into master - then from there on I can do a simpler yaml change seperatly.